Content.Server/Botany/Systems/PlantHolderSystem.cs

```diff
@@ -228,37 +228,6 @@ private void OnInteractUsing(Entity<PlantHolderComponent> entity, ref InteractUs
             return;
         }
 
-        if (_solutionContainerSystem.TryGetDrainableSolution(args.Used, out var solution, out _)
-            && _solutionContainerSystem.ResolveSolution(uid, component.SoilSolutionName, ref component.SoilSolution)
-            && TryComp(args.Used, out SprayComponent? spray))
-        {
-            var amount = FixedPoint2.New(1);
-
-            var targetEntity = uid;
-            var solutionEntity = args.Used;
-
-            _audio.PlayPvs(spray.SpraySound, args.Used, AudioParams.Default.WithVariation(0.125f));
-
-            var split = _solutionContainerSystem.Drain(solutionEntity, solution.Value, amount);
-
-            if (split.Volume == 0)
-            {
-                _popup.PopupCursor(Loc.GetString("plant-holder-component-no-plant-message",
-                    ("owner", args.Used)), args.User);
-                return;
-            }
-
-            _popup.PopupCursor(Loc.GetString("plant-holder-component-spray-message",
-                ("owner", uid),
-                ("amount", split.Volume)), args.User, PopupType.Medium);
-
-            _solutionContainerSystem.TryAddSolution(component.SoilSolution.Value, split);
-
-            ForceUpdateByExternalCause(uid, component);
```

Comment:
What happens when this line is removed? Does the sprayed solution not take effect until the next update?

Reply:
The normal solution transfer was partially handled by [this](https://github.com/space-wizards/space-station-14/blob/6654e0041199874ca6fc8644ef3f36e93efc2e2d/Content.Shared/Chemistry/EntitySystems/SolutionTransferSystem.cs#L102) line of the code. So depending on the random order in which the events were raised you would have either:

A)
- transfer 1u to plantholder
- forced Update()
- transfer 5u to plantholder
- Update() with the next timed update loop

B)
- transfer 6u to plantholder
- forced Update

If we want to keep the forced update we could add it to a subscription via SolutionTransferredEvent. That would also have the benefit of not allowing players to spam plantholder updates using an empty spray bottle and an aut clicker.

Reply:
As long as the next timed update loop happens "soon enough" I think this will be fine. I just want to make sure that this update won't stall for a long time and give players the impression that spraying stuff doesn't do anything. If you double check that this is the case I give my blessing to ship this.

Reply:
I'll give it another in-game test to make sure.

Reply:
Ok, tested it again, and with the worst case scenario of a 3 second wait time until the transfer from the spray bottle is registered does not have any noticable impact. This is the same time as when you transfer a reagent from any other container, so I would say it only makes it more consistent. Furthermore the only scenario in which you would use a spraybottle is for weedkiller to kill a plant (where it would be easier to just use a spade) or pest spray, but I think pests don't work at the moment since I have never seen the pest level increase.

Reply:
Can confirm that pests aren't actually a thing that happens currently (and haven't been as far as I can remember)